TSC-Charter.md

```diff
@@ -37,11 +37,13 @@ can be used to change a regular TSC member to a voting TSC member, or to change
 a voting TSC member to a regular TSC member.
 
 No more than one-fourth of the TSC voting members may be affiliated with the
-same employer. If change in TSC voting membership or a change of
-employment by a TSC voting member creates a situation where more than
-one-fourth of the TSC voting membership shares an employer, then the situation
-must be immediately remedied by the removal of voting member status from one or
-more TSC voting members affiliated with the over-represented employer(s).
+the same company (by any possible means, both as employment and contractual
+obligations).
+If a change in TSC voting membership or a change of employment by a TSC voting member
+creates a situation where more than one-fourth of the TSC voting membership are
+affiliated with the same company, then the situation must be immediately remedied
+by the removal of voting member status from one or more TSC voting members affiliated
+with the over-represented company.
```

Comment:
```suggestion
the same company (by any possible means, both as employment and contractual
obligations).
If a change in TSC voting membership or a change of employment by a TSC voting member
creates a situation where more than one-fourth of the TSC voting membership are
affiliated with the same company, then the situation must be immediately remedied
by the removal of voting member status from one or more TSC voting members affiliated
with the over-represented company.

Affiliation definition:
- Affiliation includes direct employment, substantial contractual obligations (e.g., over 50% of income from one company), or formal representation of a company's interests (board member, etc..). Contributions funded through general-purpose sponsorship platforms (e.g., Tidelift, GitHub Sponsors) are not automatically considered an affiliation unless they are explicitly tied to a single company's funding or interests.
- Contractors and freelancers are considered affiliated with a company only if the majority of their time or income is tied to a single company for work directly related to the project's scope.
- Individuals funded by public grants or organizations serving as fiscal hosts (e.g., Alpha Omega, STF) are not considered affiliated unless their work explicitly and exclusively represents the interests of a single company.
- TSC members must update their affiliations quarterly or upon any significant change in their employment or contractual status. A shared disclosure registry should be maintained to track affiliations transparently while protecting personal privacy.
```

I appreciate this policy change and the intent behind it, as it aims to foster diversity and balanced representation within the TSC 👏.

However, I’d like to highlight a few scenarios where the updated wording might create complexities or unintended challenges:

**1. Maintainers Funded by Sponsorship Platforms:** Developers supported through platforms like Tidelift, GitHub Sponsors, or Open Collective may now fall under the “affiliation” category if a significant portion of their income originates from a single company using these platforms. This could inadvertently include individuals whose primary contributions are to other open-source projects. Also in some cases you have to sign a service contract sharing an employer/client with the platform itself (Tidelift case)

**2. Contractors and Fiscal Hosts:** The inclusion of contractors under the expanded definition of affiliation could lead to complications. For instance, if a company acts as a fiscal host for public grants (e.g., Alpha Omega, STF) and engages several contractors for open-source contributions, all those contractors might be considered affiliated, potentially affecting the TSC's compliance with the rule. Note that Public grants/foundations are not in scope as they are non-profit and we only refer to "companies" in this policy.

**3. Broader Scope and Implications:** The broadened definition of affiliation increases the administrative burden on TSC members to maintain updated and accurate records of their affiliations. This is particularly relevant for freelancers or contractors who often work with multiple clients simultaneously, some of which may fall under the scope of this policy.

Given these broader implications, it might be beneficial to establish a mechanism for regularly updating and tracking affiliations. This could include clear guidelines on what constitutes "affiliation" in various scenarios and a streamlined process to ensure compliance while minimizing misunderstandings

By adding a sharper definition we can avoid some of this scenarios, but maybe I am overthinking here :thinking:

Reply:
Sponsorship doesn’t create contractual obligations by definition, and I’d say any time there’s money changing hands that’s tied to specific deliverables, it’s an affiliation, and github sponsors doesn’t require a specific callout. (Tidelift is a gray area where the money is paid unrelated to deliverables, but there’s a contractual expectation that certain standards should be maintained, so I’d prefer to see that not count as an affiliation)

Reply:
I'm not sure why we would need to clarify this further when the CPC project progression criteria use the exact same terms without any clarification:

> Have a defined governing body of at least 5 or more members (owners and core maintainers), of which no more than 1/3 is affiliated with the same employer. In the case there are 5 governing members, 2 may be from the same employer.

Note that this change is _necessary_ to match the requirement for impact stage https://github.com/openjs-foundation/cross-project-council/blob/main/PROJECT_PROGRESSION.md#impact-stage. Therefore, I'm not sure _why_ we are asking to clarify this while the CPC does not clarify it in the first place.

I'll propose some updated text anyway.

Reply:
Text pushed.

Reply:
In today's TSC meeting when we discussed this, I pointed out that the charter doesn't currently say _why_ we have this restriction in place, and that first stating the goal would allow less prescriptive sentences (as we'll never catch all edge cases). The updated text that @mcollina has added in https://github.com/nodejs/TSC/pull/1650/commits/45d159fbf8a6bea83e1a0ffada0b260139c726fb addresses this point.

Reply:
I think documenting the motivation should be enough - as long as TSC nomination is still based on contribution, it would be a rather distant threat anyway, I find it rather unlikely that a company would be able to invest more than 5 contributors in the project who are active enough to become TSC members, and this issue had not even come up before when we had more people contributing to the project at least as a fraction of their job, let alone now...

Reply:
Thank you for all the feedback! We also discussed this during the CPC, and I now have a clear understanding of how it is redacted and its aim. I consider this suggestion resolved 👍